sycl/test-e2e/Basic/launch_queries/max_sub_group_size.cpp

```diff
@@ -0,0 +1,86 @@
+// REQUIRES aspect-subgroup
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/detail/core.hpp>
+#include <sycl/detail/info_desc_helpers.hpp>
+#include <sycl/kernel.hpp>
+#include <sycl/kernel_bundle.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cassert>
+#include <cstdint>
+
+namespace syclex = sycl::ext::oneapi::experimental;
+using value_type = int64_t;
+
+namespace kernels {
+
+template <class T, size_t Dim>
+using sycl_global_accessor =
+    sycl::accessor<T, Dim, sycl::access::mode::read_write,
+                   sycl::access::target::global_buffer>;
+
+class TestKernel {
+public:
+  static constexpr bool HasLocalMemory{false};
+
+  TestKernel(sycl_global_accessor<value_type, 1> acc) : acc_{acc} {}
+
+  void operator()(sycl::nd_item<1> item) const {
+    const auto gtid = item.get_global_linear_id();
+    acc_[gtid] = gtid + 42;
+  }
+
+private:
+  sycl_global_accessor<value_type, 1> acc_;
+};
+} // namespace kernels
+
+int main() {
+  sycl::queue q{};
+  const auto ctx = q.get_context();
+  auto bundle = sycl::get_kernel_bundle<sycl::bundle_state::executable>(ctx);
+  auto kernel = bundle.template get_kernel<kernels::TestKernel>();
+
+  // get value to compare with
```

Comment:
Here you submit a distinct kernel, which is different from the one described by `TestKernel` functor. There is no guarantee that maximum sub-group size you get here is the same the one you will get for `TestKernel`.